code/analyticsedge/src/phone/java/com/adobe/marketing/mobile/AnalyticsExtension.java

```diff
@@ -26,10 +26,11 @@
 class AnalyticsExtension extends Extension implements EventsHandler {
 
     private ConcurrentLinkedQueue<Event> eventQueue = new ConcurrentLinkedQueue<>();
-    private PlatformServices platformServices = new AndroidPlatformServices();
+    private PlatformServices platformServices;
     private ExecutorService executorService;
     private final Object executorMutex = new Object();
     private Map<String, Object> currentConfiguration = new HashMap<>(); // the last valid config shared state
+    private SystemInfoService systemInfoService;
```

Comment:
Can we make PlatformServices and SystemInfoService variables final, if we won't change them.

Reply:
i initialize the platformService and systemInfoService in the constructor due testing constructor being added.

Reply:
yeah, we can still make them final.

Reply:
sure will update this